### PR TITLE
fix(runtime): autocapitalize property is a string

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Unit tests (`npm test`) were run locally and passed
- [x] E2E Tests (`npm run test.karma.prod`) were run locally and passed
- [x] Prettier (`npm run prettier`) was run locally and passed

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

GitHub Issue Number: N/A

I had previously created a PR in https://github.com/ionic-team/stencil/pull/2509 which changed these types to `any` as suggested by Manu. This was done to fix https://github.com/ionic-team/ionic-framework/pull/21464.

Upon further investigation I found that this fix was not correct. The Framework issue had nothing to do with the types on `HTMLStencilElement` and instead were related to the types on `HTMLElement`.

## What is the new behavior?

- The autocapitalize type is now `string`.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

Migration path: Update usage of `autocapitalize` to be a valid value (See https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/autocapitalize). This only impacts developers who are incorrectly passing non-string values to `autocomplete` (i.e. `autocomplete={true}`).

### CODE REVIEW

**Review by rwaskiewicz — APPROVED** [review]
@tanner-reits @alicewriteswrongs can one of you TAL and provide a second review for this please? I'll update my current WIP PR for the breaking changes guide to include this once its approved

**Review by alicewriteswrongs — APPROVED** [review]
possibly typing as `'off' | 'on' | 'none' | 'sentences' | 'words' | 'characters'` was going to be my only question, everything else looks good!

**Pull request merged**
